components/code_block/code_block.tsx

```diff
@@ -75,7 +75,20 @@ const CodeBlock: React.FC<Props> = ({id, code, language, searchedContent}: Props
         htmlContent = `${searchedContent} ${content}`;
     }
 
+    const [codeBlockSelection, setCodeBlockSelection] = useState('');
+
+    const setSelection = () => {
+        const selectedText = document.getSelection();
+        if (selectedText) {
+            setCodeBlockSelection(selectedText.toString());
+        }
+    };
+
     const copyText = () => {
+        if (codeBlockSelection) {
+            copyToClipboard(codeBlockSelection);
+            return;
+        }
```

Comment:
this seems easy enough to get when running the copy function.

So my suggestion would be to get rid of the component state and just use this:
```suggestion
        const codeBlockSelection = document.getSelection();
        if (codeBlockSelection) {
            copyToClipboard(codeBlockSelection);
            return;
        }
```

Reply:
So the issue I ran into when originally doing this is that when the ContextMenu is loaded, the focus (?) is changed, so `document.getSelection()` at this point returns an empty object since it's looking for a selection within the ContextMenu. I setup the component state setter to be called on the render of the ContextMenu so that it still has the textbox in focus and correctly grabs selection. 

Do you know if there's a better way to grab that selection than `document.getSelection()`? or if we could make the focus switch back to the textbox before getting the selection?

Reply:
that's certainly strange, since the selection should be handled by the `document` and since the context-menu is part of that it should not have a problem with getting it.

Nevermind, this is a small enough component to not make a huge difference even with it using component state.